File: 09095.cpp
#include <cstdio>
#define MAX 11

int ways(int number){
	int a[MAX] = {0, 1, 2, 4};
	for(int i = 4; i <= number; i++)
		a[i] = a[i-1] + a[i-2] + a[i-3];
	return a[number];
}
int main(){
	int t;
	scanf("%d", &t);
	for(int i = 0; i < t; i++){
		int n;
		scanf("%d", &n);
		printf("%d\n", ways(n));
	}
	return 0;
}

File: 09095.py
partitions = {1: 1, 2: 2, 3: 4}


def ways(number):
    if number not in partitions.keys():
        partitions[number] = ways(number-1) + ways(number-2) + ways(number-3)
    return partitions[number]


for i in range(int(input())):
    print(ways(int(input())))
